qt/aqt/table.py

```diff
@@ -0,0 +1,1135 @@
+# -*- coding: utf-8 -*-
+# Copyright: Ankitects Pty Ltd and contributors
+# License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+from __future__ import annotations
+
+import time
+from abc import ABC, abstractmethod, abstractproperty
+from dataclasses import dataclass
+from operator import itemgetter
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Generator,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+)
+
+import aqt
+import aqt.forms
+from anki.cards import Card, CardId
+from anki.collection import BrowserRow, Collection, Config, OpChanges
+from anki.consts import *
+from anki.errors import NotFoundError
+from anki.notes import Note, NoteId
+from anki.utils import ids2str, isWin
+from aqt import colors, gui_hooks
+from aqt.qt import *
+from aqt.theme import theme_manager
+from aqt.utils import (
+    TR,
+    KeyboardModifiersPressed,
+    qtMenuShortcutWorkaround,
+    restoreHeader,
+    showInfo,
+    tr,
+)
+
+Item = Union[CardId, NoteId]
+ItemList = Union[List[CardId], List[NoteId]]
+
+
+@dataclass
+class SearchContext:
+    search: str
+    browser: aqt.browser.Browser
+    order: Union[bool, str] = True
+    # if set, provided card ids will be used instead of the regular search
+    # fixme: legacy support for card_ids?
+    ids: Optional[Sequence[Item]] = None
```

Comment:
Should we try to make this backwards compatible in card state? I guess we could maintain card_ids as an alias for ids and ignore it note state, but we would still need a new hook to prevent old add-ons from messing up order.

Reply:
A grep of the add-ons seems to indicate none are using the .card_ids property. I vote for just removing it entirely until demand presents itself